kernel/nic_queues/src/lib.rs

```diff
@@ -18,23 +20,55 @@ use alloc::{
     vec::Vec,
     collections::VecDeque
 };
-use memory::MappedPages;
+use memory::{MappedPages, create_contiguous_mapping, EntryFlags};
 use intel_ethernet::descriptors::{RxDescriptor, TxDescriptor};
-use nic_buffers::{ReceiveBuffer, ReceivedFrame};
+use nic_buffers::{ReceiveBuffer, ReceivedFrame, TransmitBuffer};
 
+/// The mapping flags used for pages that the NIC will map.
+pub const NIC_MAPPING_FLAGS: EntryFlags = EntryFlags::from_bits_truncate(
+    EntryFlags::PRESENT.bits() |
+    EntryFlags::WRITABLE.bits() |
+    EntryFlags::NO_CACHE.bits() |
+    EntryFlags::NO_EXECUTE.bits()
+);
+
+/// The register trait that gives access to only those registers required for receiving a packet.
+/// The Rx queue control registers can only be accessed by the physical NIC.
+pub trait RxQueueRegisters {
+    fn update_rdbal(&mut self, value: u32);
+    fn update_rdbah(&mut self, value: u32);
+    fn update_rdlen(&mut self, value: u32);
+    fn update_rdh(&mut self, value: u32);
+    fn update_rdt(&mut self, value: u32);
```

Comment:
Nit:  is `update_foo()` the right naming convention? Do NIC manuals use that? Typically Rust uses things like `set_` or `write_`.

Reply:
more just how I think of writing to a register as "updating" it. I'll change them to set